### PR TITLE
feat(scalar): add phone no

### PR DESCRIPTION
**Pull request by ssddOnTop**
**Issue Reference(s):**  
https://github.com/tailcallhq/tailcall/issues/1214

TODO:

- [x] add generic test for all scalars

## Summary by CodeRabbit

- **Refactor**
	- Improved email validation with a more comprehensive regular expression pattern.
- **New Features**
	- Added a new scalar type `PhoneNumber` to the schema definition.
	- Expanded functionality to include a new custom scalar `PhoneNumber`.
	- Introduced a `PhoneNumber` struct for phone numbers in E.164 format with validation logic.
- **Tests**
	- Updated test cases to include the new `PhoneNumber` field with invalid inputs.

### CODE REVIEW

File: generated/.tailcallrc.schema.json [hunk 1]
```diff
@@ -1365,6 +1365,18 @@
...
+      "type": "object",
+      "required": [
+        "PhoneNumber"
+      ],
+      "properties": {
+        "PhoneNumber": {
+          "description": "A field whose value conforms to the standard E.164 format as specified in E.164 specification (https://en.wikipedia.org/wiki/E.164).",
+          "type": "string"
+        }
+      }
```
[the remark this answers is not in the document]
Comment (ssddOnTop) [reply generated/.tailcallrc.schema.json:1378]: the crate is widely used and it's covered with tests, marking this as resolved.

File: tests/execution/test-scalars.md [hunk 2]
```diff
@@ -21,4 +23,8 @@ type Query {
   url: http://localhost:8000/graphql
   body:
     query: '{ value(email: "alo@valid.com") }'
+- method: POST
+  url: http://localhost:8000/graphql
+  body:
+    query: '{ value_phone(phone: "+1 (614) 1234567") }'
```
[the remark this answers is not in the document]
Comment (ssddOnTop) [reply tests/execution/test-scalars.md:29]: this is intentional. Marking this as resolved.

File: tests/snapshots/execution_spec__test-scalars-invalid.md_merged.snap [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+input Email {
+
+}
+
+input PhoneNumber {
+
+}
+
```
Comment (amitksingh1490) [inline tests/snapshots/execution_spec__test-scalars-invalid.md_merged.snap:16]: shouldn't this come as scalar?

File: tests/snapshots/execution_spec__test-scalars.md_merged.snap [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+input Email {
+
+}
+
+input PhoneNumber {
+
+}
+
```
Comment (amitksingh1490) [inline tests/snapshots/execution_spec__test-scalars.md_merged.snap:16]: this should come as scalar

Comment (ssddOnTop) [reply tests/snapshots/execution_spec__test-scalars.md_merged.snap:16]: it's the generated outpur from config.to_sdl().. I don't think it's related to this pr

File: tests/execution/test-scalars.md [hunk 5]
```diff
@@ -4,13 +4,15 @@
...
 scalar Email
+scalar PhoneNumber
 
 schema @server(port: 8000, graphiql: true, hostname: "localhost") {
   query: Query
 }
 
 type Query {
   value(email: Email!): Email! @const(data: "{{args.email}}")
+  value_phone(phone: PhoneNumber!): PhoneNumber! @const(data: "{{args.phone}}")
```
Comment (tusharmath) [inline tests/execution/test-scalars.md:15]:
```suggestion
  email(value: Email!): Email! @const(data: "{{args.email}}")
  phone(value: PhoneNumber!): PhoneNumber! @const(data: "{{args.phone}}")
```

File: tests/execution/test-scalars-invalid.md [hunk 6]
```diff
@@ -4,13 +4,15 @@
...
 scalar Email
+scalar PhoneNumber
 
 schema @server(port: 8000, graphiql: true, hostname: "localhost") {
   query: Query
 }
 
 type Query {
   value(email: Email!): Email! @const(data: "{{args.email}}")
+  value_phone(phone: PhoneNumber!): PhoneNumber! @const(data: "{{args.phone}}")
```
Comment (tusharmath) [inline tests/execution/test-scalars-invalid.md:15]:
```suggestion
  email(value: Email!): Email! @const(data: "{{args.email}}")
  phone(value: PhoneNumber!): PhoneNumber! @const(data: "{{args.phone}}")
```

File: tests/execution/test-scalars-invalid.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  body:
+    query: '{ email(value: "alo@invalid") }'
+- method: POST
+  url: http://localhost:8000/graphql
+  body:
+    query: '{ phone(value: "0") }'
+- method: POST
+  url: http://localhost:8000/graphql
+  body:
+    query: '{ phone(value: "1234567890") }'
```
[the remark this answers is not in the document]
Comment (ssddOnTop) [reply tests/execution/test-scalars-invalid.md:33]: these are covered under the library, marking it as resolved

**Review by tusharmath — APPROVED** [review]

**Pull request merged**